common/djangoapps/request_cache/middleware.py

```diff
@@ -58,3 +58,50 @@ def process_exception(self, request, exception):  # pylint: disable=unused-argum
         """
         self.clear_request_cache()
         return None
+
+
+def request_cached(wrapped):
+    """
+    Wraps a function and automatically handles caching its return value, as well as
+    returning that cached value for subsequent calls to the same function, with the
+    same parameters, within a given request.
+
+    Caveats:
+        - we convert positional arguments to their string equivalent and use them in the cache key, so if something
+          can't be converted to a string, things might blow up
+        - our cache key depends on the positional args (so , holistically, cache efficiency is tied to cardinality)
+        - we use the default request cache, not a named request cache
+        - the cache key is prefixed with the name of the wrapped function, so theoretically, an identically named
+          function elsewhere in the code with the same parameters could overwrite cached values
+    """
+    def wrapper(*args, **kwargs):
+        """
+        Wrapper function to decorate with.
+        """
+
+        # Build our cache key based on the name of the function we're wrapping
+        # and the parameters passed in to it.  This could get a little dicey
+        # but the intention is that the only functions being decorated with this
+        # are akin to pure functions, where you're passing in simple values and
+        # getting something out, and don't require full objects that you modify,
+        # or anything like that.
+        converted_params = map(str, args)
```

Comment:
Can you add `wrapped.__module__` to the key to help with name spacing?  This is probably in place of #2 above.  This would be closer to the code that you are replacing.